libs/librepcb/projecteditor/boardeditor/fsm/bes_select.cpp

```diff
@@ -759,6 +771,91 @@ bool BES_Select::removeSelectedItems() noexcept {
   }
 }
 
+void BES_Select::measureLengthInDirection(bool              directionBackwards,
+                                          const BI_NetLine& netline,
+                                          QSet<Uuid>&       visitedSegments,
+                                          UnsignedLength&   totalLength) {
+  auto currentNetline = std::cref(netline);
+  auto currentAnchor  = std::cref(directionBackwards ? netline.getStartPoint()
+                                                    : netline.getEndPoint());
+
+  for (;;) {
+    tl::optional<BI_NetLine&> nextNetline = tl::nullopt;
+    for (BI_NetLine* nl : currentAnchor.get().getNetLines()) {
+      // Don't visit a segment twice
+      if (visitedSegments.contains(nl->getUuid())) {
+        continue;
+      }
+      // Only visit selected segments
+      if (nl->isSelected()) {
+        if (nextNetline.has_value()) {
+          // There's already another connected and selected netline
+          throw LogicError(__FILE__, __LINE__,
+                           tr("Selected segment may not have branches"));
+        }
+
+        totalLength += nl->getLength();
+        nextNetline = *nl;
+        visitedSegments.insert(nl->getUuid());
+      }
+    }
+    if (nextNetline.has_value()) {
+      currentNetline = std::cref(*nextNetline);
+      currentAnchor =
+          std::cref(*nextNetline->getOtherPoint(currentAnchor.get()));
+    } else {
+      break;
+    }
+  }
+}
+
+bool BES_Select::measureSelectedItems(BI_NetLine& netline) noexcept {
+  Board* board = mEditor.getActiveBoard();
+  Q_ASSERT(board);
+  if (!board) return false;
+
+  Q_ASSERT(netline.isSelected());
+
+  // Store UUIDs of visited segments
+  QSet<Uuid> visitedSegments;
```

Comment:
For such use-cases I usually use a set of pointers (`QSet<const BI_NetLine*>`) since a pointer is even "more unique" than a UUID and probably also more efficient :wink: However, `QSet<Uuid>` is also fine for me...

But I'd really rename the variable since it doesn't contain segments (in the meaning of `BI_NetSegment`), but netlines:

```suggestion
  QSet<Uuid> visitedNetLines;
```

Reply:
Good point, I'll rename.

In theory there could be two pointers to the same element, right? Using UUIDs seems safer and any performance difference should be negligible.

Reply:
> In theory there could be two pointers to the same element, right?

Yes there could be multiple instances (and thus different pointers) of the same netline.

> Using UUIDs seems safer

Generally yes, but in this case it doesn't really matter since you wouldn't dereference these pointers - the pointers would just be a unique identification of the netlines.

Anyway, I'm fine with `QSet<Uuid>`.